src/components/encoding-pane/filter-pane.tsx

```diff
@@ -0,0 +1,226 @@
+import {FieldQuery} from 'compassql/build/src/query/encoding';
+import {ExpandedType} from 'compassql/build/src/query/expandedtype';
+import {Schema} from 'compassql/build/src/schema';
+import {isWildcard} from 'compassql/build/src/wildcard';
+import * as React from 'react';
+import * as CSSModules from 'react-css-modules';
+import {ConnectDropTarget, DropTarget, DropTargetCollector, DropTargetSpec} from 'react-dnd';
+import {OneOfFilter, RangeFilter} from 'vega-lite/build/src/filter';
+import {TimeUnit} from 'vega-lite/build/src/timeunit';
+import {FILTER_ADD, FILTER_MODIFY_TIME_UNIT,
+  FILTER_REMOVE, FilterAction} from '../../actions/filter';
+import {DraggableType} from '../../constants';
+import {ShelfFieldDef} from '../../models/shelf/encoding';
+import {getDefaultList, getDefaultRange} from '../../reducers/shelf/filter';
+import {DraggedFieldIdentifier} from '../field';
+import {Field} from '../field/index';
+import * as styles from './filter-pane.scss';
+import {FunctionPicker} from './function-picker';
+import {OneOfFilterShelf} from './one-of-filter-shelf';
+import {RangeFilterShelf} from './range-filter-shelf';
+
+
+
+/**
+ * Props for react-dnd of FilterShelf
+ */
+export interface FilterPaneDropTargetProps {
+  connectDropTarget: ConnectDropTarget;
+
+  isOver: boolean;
+
+  item: Object;
+}
+
+export interface FilterPanePropsBase {
+  filters: Array<RangeFilter | OneOfFilter>;
+  fieldDefs: ShelfFieldDef[];
```

Comment:
Because I need the fieldDef field and type to construct an filedQuery. This can be removed after the domain function in compassql is modified.